athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/NeptuneRecordHandler.java

```diff
@@ -0,0 +1,248 @@
+     *
+     * @param spiller            A BlockSpiller that should be used to write the row
+     *                           data associated with this Split. The BlockSpiller
+     *                           automatically handles chunking the response,
+     *                           encrypting, and spilling to S3.
+     * @param recordsRequest     Details of the read request, including: 1. The
+     *                           Split 2. The Catalog, Database, and Table the read
+     *                           request is for. 3. The filtering predicate (if any)
+     *                           4. The columns required for projection.
+     * @param queryStatusChecker A QueryStatusChecker that you can use to stop doing
+     *                           work for a query that has already terminated
+     * @throws IOException
+     * @note Avoid writing >10 rows per-call to BlockSpiller.writeRow(...) because
+     *       this will limit the BlockSpiller's ability to control Block size. The
+     *       resulting increase in Block size may cause failures and reduced
+     *       performance.
+     */
+    @Override
+    protected void readWithConstraint(final BlockSpiller spiller, final ReadRecordsRequest recordsRequest,
+            final QueryStatusChecker queryStatusChecker) {
+        logger.info("readWithConstraint: enter - " + recordsRequest.getSplit());
+        final TableName tableName = recordsRequest.getTableName();
+        final String labelName = tableName.getTableName();
+        long numRows = 0;
+        final AtomicLong numResultRows = new AtomicLong(0);
+        final Client client = neptuneConnection.getNeptunClientConnection();
+
+        try {
+            final GraphTraversalSource graphTraversalSource = neptuneConnection.getTraversalSource(client);
+
+            String traversal = "g.V().hasLabel('" + labelName + "')";
+
+            if (recordsRequest.getConstraints().getSummary().size() > 0) {
+
+                try {
+                    logger.info("readWithContraint: Constaints Map "
+                            + recordsRequest.getConstraints().getSummary().toString());
+
+                    final Map<String, ValueSet> constraints = recordsRequest.getConstraints().getSummary();
+                    traversal += flattenContraintsMap(constraints);
+                } catch (final Exception e) {
+                    logger.info("readWithContraint: Exception occured at typecasting " + e.getStackTrace());
+                }
+            }
+
+            traversal += ".valueMap()";
+
+            logger.info("readWithContraint: Neptune Query " + traversal);
+
+            // Code to build gremlin query from string
+            final ConcurrentBindings b = new ConcurrentBindings();
+            b.putIfAbsent("g", graphTraversalSource);
+
+            final GremlinExecutor ge = GremlinExecutor.build().evaluationTimeout(15000L).globalBindings(b).create();
```

Comment:
Can we make the 15000L configurable via env variable?

Also, what exactly is this timeout used for?

Reply:
Need to investigate

Reply:
refactored, no longer using these classes